src/connection/activate.ts

```diff
@@ -0,0 +1,94 @@
+import { sendAnalyticsEvent } from '@uniswap/analytics'
+import { InterfaceEventName, WalletConnectionResult } from '@uniswap/analytics-events'
+import { sendEvent } from 'components/analytics'
+import { Connection } from 'connection'
+import { atom, useAtom } from 'jotai'
+import { useCallback } from 'react'
+import { useAppDispatch } from 'state/hooks'
+import { updateSelectedWallet } from 'state/user/reducer'
+
+import { didUserReject } from './utils'
+
+type PendingConnectionState = {
+  connection: Connection | undefined
+  error: any
+}
+
+const pendingConnectionAtom = atom<PendingConnectionState>({ connection: undefined, error: undefined })
+
+export function useActivateConnection() {
+  const dispatch = useAppDispatch()
+
+  const [pending, setPending] = useAtom(pendingConnectionAtom)
+
+  const updatePendingConnection = useCallback(
+    (connection: Connection | undefined) => {
+      // log selected wallet
+      if (connection !== undefined) {
```

Comment:
I know we have moved this code over from previous implementation, but I just wanted to ask whether logging an event here makes sense. This occurs while connection is pending. Not while connection was completed. In case it errors, we will send an event of changed wallet, but that did not complete successfully.

Reply:
I'm not sure if it makes sense either. I noticed that as well when I was moving it over, what do you think we should do here?

Reply:
Turns out we no longer need this google-analytics event! Removed 👍